### PR TITLE
Closes #2113. Fixes bug in test and updates testnet fr addresses.

### LINKED ISSUE #2113: Upgrade testnet to fix bug in test and update fr addresses
Due to a bug in a unit test used to create addresses, the testnet fr addresses did not have their corresponding keys retained.

Address 1 t2UNzUUx8mWBCRYPRezvA363EYXyEpHokyi was used between blocks 1-17708
Address 2 t2N9PH9Wk9xjqYg9iin1Ua3aekJqfAtE543 was used between blocks 17709 - 35417 (count of 17709)
Address 3 t2NGQjYMQhFndDHguvUw4wZdNdsssA6K7x2 is being used for blocks 35418 - 53126 (count of 17709)
Address 4 will be used from block 53127 (based on historical rate of blocks, estimated to arrive around 20th March 2017)

Testnet is currently around block 42650 - see https://explorer.testnet.z.cash/

This PR fixes the bug and introduces new addresses 4-48.  This will change consensus so all testnet nodes are required to update.  To be more conservative so users have more time to upgrade their testnet nodes, we could start with address 5.

If this PR is merged for 1.0.7,
- testnet.z.cash will be upgraded to 1.0.7
- testnet alert will be sent, putting testnet nodes running 1.0.0-1.0.6 into safe mode (wallet rpc commands will be disabled), to encourage users to upgrade

### PR DESCRIPTION
**Pull request by bitcartel**
Updating the testnet fr addresses will introduce a consensus change on testnet, so all testnet nodes will be required to upgrade.  See #2113 for more information.

### CODE REVIEW

File: src/gtest/test_foundersreward.cpp [hunk 1]
```diff
@@ -101,11 +101,11 @@ TEST(founders_reward_test, general) {
     CChainParams params = Params();
     
     // First testnet reward:
-    // address = t2UNzUUx8mWBCRYPRezvA363EYXyEpHokyi
-    // script = OP_HASH160 ef775f1f997f122a062fff1a2d7443abd1f9c642 OP_EQUAL
-    // raw script = a914ef775f1f997f122a062fff1a2d7443abd1f9c64287
-    EXPECT_EQ(params.GetFoundersRewardScriptAtHeight(1), ParseHex("a914ef775f1f997f122a062fff1a2d7443abd1f9c64287"));
-    EXPECT_EQ(params.GetFoundersRewardAddressAtHeight(1), "t2UNzUUx8mWBCRYPRezvA363EYXyEpHokyi");
+    // address = t2Mcc7khmWtviJKZLn9XyqPcs2zfQDn5K4X
```
Comment (daira) [inline src/gtest/test_foundersreward.cpp]: This is no longer correct because of 20a95de93e192129117ce75e9a82a9001d36126e .

Comment (daira) [inline src/gtest/test_foundersreward.cpp]: I suggest using ``t2ENg7hHVqqs9JwU5cgjvSbxnT2a9USNfhy`` and ``GetFoundersReward{Script,Address}AtHeight(53127)``.

Comment (bitcartel) [inline src/gtest/test_foundersreward.cpp]: Done.  Added check for 53126 and 53127 to confirm addresses change at that point. Force pushed to keep commits clean.

**Comment by bitcartel** [comment]
@daira Changes made per review.

**Comment by daira** [comment]
@zkbot r+

**Pull request merged**